### PR TITLE
pyln.proto.message minor fixes

### PR DESCRIPTION
**Pull request by rustyrussell**
My tests weren't exhaustive, and now I'm hitting things in more complex testing of the spec.
Changelog-None

### CODE REVIEW

File: contrib/pyln-proto/pyln/proto/message/array_types.py [hunk 1]
```diff
@@ -41,6 +41,13 @@ def val_to_str(self, v: List[Any], otherfields: Dict[str, Any]) -> str:
         s = ','.join(self.elemtype.val_to_str(i, otherfields) for i in v)
         return '[' + s + ']'
 
+    def val_to_py(self, v: Any, otherfields: Dict[str, Any]) -> Union[str, List[Any]]:
```
Comment (cdecker) [inline contrib/pyln-proto/pyln/proto/message/array_types.py:44]: Why is the `to_py()` method not part of the instances themselves, but rather the type class? Seems like a weird indirection.

Comment (cdecker) [reply contrib/pyln-proto/pyln/proto/message/array_types.py:44]: It'd also mean that the instance type (type of `v`) can extend the native python type and just work as one, without having to have an explicit conversion.

**Comment by rustyrussell** [comment]
It's because we create a class instance representing a bolt type.  It would be cleaner to make it a real type, then dynamically generate the other classes, but that's a bit advanced for me.

And in practice it doesn't make a huge difference to class users, who rarely interact below the Message level.

I would like to rewrite it, but time....

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3770/commits/600b56eca47a5185abddf2ad2f6fd83d8bb34f69

**Pull request merged**